crypto/param_build.c

```diff
@@ -144,18 +136,23 @@ int ossl_param_bld_push_double(OSSL_PARAM_BLD *bld, const char *key,
 int ossl_param_bld_push_BN(OSSL_PARAM_BLD *bld, const char *key,
                            const BIGNUM *bn)
 {
-    int sz = BN_num_bytes(bn);
+    int sz = -1, secure = 0;
     OSSL_PARAM_BLD_DEF *pd;
 
-    if (sz <= 0) {
-        CRYPTOerr(CRYPTO_F_OSSL_PARAM_BLD_PUSH_BN, CRYPTO_R_ZERO_LENGTH_NUMBER);
-        return 0;
+    if (bn != NULL) {
+        sz = BN_num_bytes(bn);
+        if (sz <= 0) {
+            CRYPTOerr(CRYPTO_F_OSSL_PARAM_BLD_PUSH_BN,
+                      CRYPTO_R_ZERO_LENGTH_NUMBER);
+            return 0;
+        }
```

Comment:
That means we can't pass zero.  `BN_num_bytes` returns zero for the number zero.

Reply:
No algorithm can have a zero parameter :)

Reply:
Well...

Reply:
Actually, yes they can.  We don't usually expect a BIGNUM to be used for them, but...
(think "flags")

Reply:
I slipped in the fix since the comment was on the line after, it didn't register :)